doc/releases/release-notes-2.3.rst

```diff
@@ -51,6 +51,11 @@ Removed APIs in this release
 Stable API changes in this release
 ==================================
 
+* Bluetooth Mesh
+
+  * Health client model API has been reworded to deteted `net_idx` parameter.
+  Since we no longer need this parameter, it can be derived from app_idx.
```

Comment:
I'd suggest changing this to: "The `net_idx` parameter has been removed from the Health Client model APIs since it can be derived (by the stack) from the `app_idx` parameter".

Also, I think the preference is to have release notes changes in independent commits. @carlescufi is that correct?